apps/ewallet/lib/ewallet/gates/blockchain_transaction_gate.ex

```diff
@@ -0,0 +1,190 @@
+# Copyright 2018-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EWallet.BlockchainTransactionGate do
+  @moduledoc """
+  Handles the logic for blockchain transactions. Validates the inputs, inserts the
+  initial transaction before calling a Transaction Tracker that will take care
+  of updating the fields based on events coming from the blockchain side.
+
+  This is using a different approach than the LocalTransactionGate
+  for simplicity (modifying the existing inputs instead of creating new
+  data structures).
+  """
+  alias EWallet.{
+    BlockchainTransactionPolicy,
+    TokenFetcher,
+    BlockchainTransactionState,
+    TransactionRegistry,
+    TransactionTracker
+  }
+
+  alias EWalletDB.{BlockchainWallet, Transaction}
+  alias ActivityLogger.System
+
+  # TODO: Check if blockchain is enabled
+  # TODO: Check if blockchain is available
+  # TODO: Add tests at the controller level
+  # TODO: Add tests for failures at the gate level
+
+  # Here, we send a transaction from the hot wallet to
```

Comment:
y u no like `@doc`

Reply:
Sorry, I'm new to this whole thing, used to work exclusively on the admin panel.

Reply:
🙄